secret_store/src/key_server_cluster/cluster.rs

```diff
@@ -293,53 +290,49 @@ impl ClusterCore {
 
 	/// Connect to peer.
 	fn connect(data: Arc<ClusterData>, node_address: SocketAddr) {
-		data.handle.clone().spawn(move |handle| {
-			data.pool.clone().spawn(ClusterCore::connect_future(handle, data, node_address))
-		})
+		data.clone().spawn(ClusterCore::connect_future(data, node_address));
 	}
 
-	/// Connect to socket using given context and handle.
-	fn connect_future(handle: &Handle, data: Arc<ClusterData>, node_address: SocketAddr) -> BoxedEmptyFuture {
+	/// Connect to socket using given context and executor.
+	fn connect_future(data: Arc<ClusterData>, node_address: SocketAddr) -> BoxedEmptyFuture {
 		let disconnected_nodes = data.connections.disconnected_nodes().keys().cloned().collect();
-		Box::new(net_connect(&node_address, handle, data.self_key_pair.clone(), disconnected_nodes)
+		Box::new(net_connect(&node_address, data.self_key_pair.clone(), disconnected_nodes)
 			.then(move |result| ClusterCore::process_connection_result(data, Some(node_address), result))
-			.then(|_| finished(())))
+			.then(|_| future::ok(())))
 	}
 
 	/// Start listening for incoming connections.
-	fn listen(handle: &Handle, data: Arc<ClusterData>, listen_address: SocketAddr) -> Result<BoxedEmptyFuture, Error> {
-		Ok(Box::new(TcpListener::bind(&listen_address, &handle)?
+	fn listen(data: Arc<ClusterData>, listen_address: SocketAddr) -> Result<BoxedEmptyFuture, Error> {
```

Comment:
I haven't traced through all the calls but aren't we dropping a few possible errors in this method? Perhaps out of scope for this PR though...

Reply:
Yes, the errors are not handled the way I would do it. I've tried not to change too much functionality from the original in this PR in order to keep things as straightforward as possible. I'll make a note to have a look at that in an upcoming PR :)

Reply:
@dvdplm Not sure I got it - all connection-establish (both for incoming && outcoming) related errors are processed in `process_connection_result` by tracing? Probably some other errors are dropped? Or do you have idea how to handle these errors in other way? Could you please, elaborate?